server/elm/src/DateSelector/Selector.elm

```diff
@@ -0,0 +1,418 @@
+    let
+        isInvertedMinMax =
+            Date.compare minimum maximum == GT
+
+        months =
+            if isInvertedMinMax then
+                []
+
+            else
+                let
+                    first =
+                        if year selectedDate == year minimum then
+                            Date.monthNumber minimum
+
+                        else
+                            1
+
+                    last =
+                        if year selectedDate == year maximum then
+                            Date.monthNumber maximum
+
+                        else
+                            12
+                in
+                List.filter
+                    (\month ->
+                        let
+                            monthNumber =
+                                Date.monthToNumber month
+                        in
+                        monthNumber >= first && monthNumber <= last
+                    )
+                    allMonths
+
+        options =
+            List.map
+                (\month ->
+                    let
+                        monthNumber =
+                            Date.monthToNumber month
+                    in
+                    option
+                        [ value <| String.fromInt monthNumber
+                        , selected <| Date.monthNumber selectedDate == monthNumber
+                        ]
+                        [ text <| translate language <| Translate.ResolveMonth False month ]
+                )
+                months
+    in
+    select
+        [ on "input" <|
+            Json.Decode.map
+                (\s ->
+                    let
+                        selectedMonths =
+                            String.toInt s
+                                |> -- We'll never get here, as we place only
+                                   -- valid numeric values at select options.
+                                   Maybe.withDefault 1
+                    in
+                    numberToMonth selectedMonths
+                        |> dateWithMonth selectedDate
+                )
+                (Json.Decode.at [ "target", "value" ] Json.Decode.string)
+        ]
+        options
+
+
+viewMonthSelectListDisabled : Html a
+viewMonthSelectListDisabled =
+    select [] [ option [ value "" ] [ text "" ] ]
+
+
+groupsOf : Int -> List a -> List (List a)
+groupsOf n list =
+    if List.isEmpty list then
+        []
+
+    else
+        List.take n list :: groupsOf n (List.drop n list)
+
+
+monthDates : Int -> Month -> List Date
+monthDates y m =
+    let
+        start =
+            Date.floor Monday <| Date.fromCalendarDate y m 1
+    in
+    Date.range Day 1 start <| Date.add Days 42 start
+
+
+dateWithYear : Date -> Int -> Date
+dateWithYear date y =
+    let
+        m =
+            month date
+
+        d =
+            day date
+    in
+    if m == Feb && d == 29 && not (isLeapYear y) then
```

Comment:
When the date is February 29 and the target year is not a leap year, it correctly changes to February 28. However, if the original date is February 28 and the target year is a leap year, it should stay February 28.

Reply:
Same as https://github.com/TIP-Global-Health/eheza-app/pull/1177#discussion_r1646641701